src/main/java/io/supertokens/inmemorydb/queries/PasswordlessQueries.java

```diff
@@ -139,23 +139,42 @@ public static void deleteDevice_Transaction(Start start, Connection con, String
 
     public static void deleteDevicesByPhoneNumber_Transaction(Start start, Connection con, @Nonnull String phoneNumber)
             throws SQLException {
-        String QUERY = "DELETE FROM " + Config.getConfig(start).getPasswordlessDevicesTable()
+        // SQLite is not compiled with foreign key constraint and so we must implement cascading deletes here
+        String QUERY = "SELECT device_id_hash FROM " + Config.getConfig(start).getPasswordlessDevicesTable()
                 + " WHERE phone_number = ?";
 
+        List<String> deviceIdHashes = new ArrayList<>();
         try (PreparedStatement pst = con.prepareStatement(QUERY)) {
             pst.setString(1, phoneNumber);
-            pst.executeUpdate();
+            ResultSet result = pst.executeQuery();
+            while (result.next()) {
+                deviceIdHashes.add(result.getString("device_id_hash"));
+            }
         }
+        deleteRowsByDeviceIdHashList_Transaction(con, Config.getConfig(start).getPasswordlessCodesTable(),
+                deviceIdHashes);
+        deleteRowsByDeviceIdHashList_Transaction(con, Config.getConfig(start).getPasswordlessDevicesTable(),
+                deviceIdHashes);
     }
```

Comment:
Why are we doing it this way? Why can't we just do a `DELETE from devices where phoneNumber = ..` (and this will also delete from the codes table automatically)? Cause If we do it this way, then we need to make the SELECT query a `FOR_UPDATE` one?

Reply:
I went with locking the entire table - it simplifies implementation, and it shouldn't matter much in trial&dev.

> creating a new device + code

I don't think we need to lock the table when creating a device with code. Creating the code and device is done in a transaction, and there is no pre-condition for creating the device depending on anything else in the DB.

> creating a new code based on a device id hash
> updating info in the devices table

Locks using `getDevice_Transaction`

Reply:
> I don't think we need to lock the table when creating a device with code. Creating the code and device is done in a transaction, and there is no pre-condition for creating the device depending on anything else in the DB.

yes. For creating a new device and a code, no locking is required. But for creating a new code, given an existing device, we would need to lock the devices table?